theano/configdefaults.py

```diff
@@ -1116,6 +1116,18 @@ def is_valid_check_preallocated_output_param(param):
              # Added elsewhere in the c key only when needed.
              in_c_key=False)
 
+if not config.cxx == "gcc" and config.gcc.cxxflags:
```

Comment:
I think we should have a setting for cxx, and another for cxx_path so that the first one can be the compiler "class" and the other one the binary path.  So you could have `config.cxx = 'g++'` and `config.cxx_path = '/usr/bin/clang++'`, for example.

This will avoid mistakes in checks like these.